### PR TITLE
Release Hermes v1.9.0

### PR DESCRIPTION
**Pull request by ljoss17**
Closes: #4015

- [x] Bump all crate versions to the new version
- [x] Update `Cargo.lock`
- [x] Create a new release in the changelog, using [`unclog`](https://github.com/informalsystems/unclog)
  - If doing a release candidate (`rc`) version, then skip the `unclog release` step
- [x] Reassign unfinished issues of previous milestone to the next milestone
- [x] Notify the comms team about the pending new release by tagging [@isa-belch](https://github.com/isa-belch) as a reviewer here

### CODE REVIEW

File: crates/relayer-cli/Cargo.toml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+version    = "1.9.0"
 edition    = "2021"
 license    = "Apache-2.0"
 readme     = "README.md"
 keywords   = ["blockchain", "consensus", "cosmos", "ibc", "tendermint"]
 homepage   = "https://hermes.informal.systems/"
 repository = "https://github.com/informalsystems/hermes"
 authors    = ["Informal Systems <hello@informal.systems>"]
-rust-version = "1.71.1"
+rust-version = "1.76.0"
```
Comment (ljoss17) [inline crates/relayer-cli/Cargo.toml:11]: MSRV with `v1.76.0`:
```
cargo msrv verify --log-level trace --manifest-path crates/relayer-cli/Cargo.toml
  [Meta]   cargo-msrv 0.16.0-beta.20                                                                                                                                                                                             

Compatibility Check #1: Rust 1.76.0
  [OK]     Is compatible
```

MSRV with `v1.75.0`:

```
cargo msrv verify --log-level trace --manifest-path crates/relayer-cli/Cargo.toml
  [Meta]   cargo-msrv 0.16.0-beta.20                                                                                                                                                                                             

Compatibility Check #1: Rust 1.75.0
  [FAIL]   Is incompatible 
                                                                                                              
  ╭──────────────────────────────────────────────────────────────────────────────────────────────────────────╮
  │     Blocking waiting for file lock on package cache                                                      │
  │     Blocking waiting for file lock on build directory                                                    │
  │     Checking tendermint-light-client-verifier v0.36.0                                                    │
  │     Checking tonic v0.11.0                                                                               │
  │     Checking tendermint-config v0.36.0                                                                   │
  │     Checking adler v1.0.2                                                                                │
  │     Checking gimli v0.28.1                                                                               │
  │     Checking os_str_bytes v6.6.1                                                                         │
  │    Compiling tiny-keccak v2.0.2                                                                          │
  │     Checking triomphe v0.1.12                                                                            │
  │ error[E0658]: use of unstable library feature 'ptr_addr_eq'                                              │
  │    --> /Users/luca/.cargo/registry/src/index.crates.io-6f17d22bba15001f/triomphe-0.1.12/src/arc.rs:282:9 │
  │     |                                                                                                    │
  │ 282 |         ptr::addr_eq(this.ptr(), other.ptr())                                                      │
  │     |         ^^^^^^^^^^^^                                                                               │
  │     |                                                                                                    │
  │     = note: see issue #116324 <https://github.com/rust-lang/rust/issues/116324> for more information     │
  │                                                                                                          │
  │ For more information about this error, try `rustc --explain E0658`.                                      │
  │    Compiling bitcoin v0.31.2                                                                             │
  │ error: could not compile `triomphe` (lib) due to previous error                                          │
  │ warning: build failed, waiting for other jobs to finish...                                               │
  │                                                                                                          │
  ╰──────────────────────────────────────────────────────────────────────────────────────────────────────────╯
                                                                                                              


Crate source was found to be incompatible with Rust version '1.75.0' specified as MSRV in the Cargo manifest located at 'crates/relayer-cli/Cargo.toml' 
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,78 @@
...
+Additionally, this release includes various bug fixes enhancing the stability and performance of Hermes. These fixes address issues with channel and connection creation on unsupported ibc-go versions, event extraction, health-check messages, and more.
+
+### BREAKING CHANGES
+
+- [Telemetry & Metrics](telemetry)
+  - Remove the `telemetry` and `rest-server` feature flags, ensuring Hermes is always built with telemetry and REST support.
+    Both servers can still be disabled in the configuration file, by setting `telemetry.enabled = false` and `rest.enabled = false`, respectively.
+    ([\#3878](https://github.com/informalsystems/hermes/pull/3878))
+
+### BUG FIXES
```
Comment (romac) [inline CHANGELOG.md:24]: Let's move this section after the features

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,78 @@
...
+
+This v1.9.0 release introduces new features and improvements to Hermes.
+
+**Major Features**:
+
+1. Channel Upgrade Compatibility: Hermes now supports channel upgrade events introduced in ibc-go v8, ensuring smoother transitions and enhanced functionality.
+
+3. Dynamic Gas Fees Compatibility: Hermes is now compatible with Skip's feemarket for dynamic gas fees, in addition to Osmosis' implementation, providing more flexibility in gas fee management.
+
+Additionally, this release includes various bug fixes enhancing the stability and performance of Hermes. These fixes address issues with channel and connection creation on unsupported ibc-go versions, event extraction, health-check messages, and more.
```
Comment (romac) [inline CHANGELOG.md:15]:
```suggestion
1. **Channel Upgrades:** Hermes now handles [channel upgrade](https://www.ibcprotocol.dev/blog/introducing-ibc-channel-upgradability) events introduced in ibc-go v8, helping chains opting-in to new functionality on existing channels.
2. **Dynamic Gas Fees Compatibility:** Hermes is now compatible with Skip's `x/feemarket` module for dynamic gas fees, in addition to Osmosis' implementation, providing more flexibility in gas fee management.

Additionally, this release includes various bug fixes enhancing the stability and performance of Hermes. These fixes address issues with channel and connection creation on older ibc-go versions, event extraction, health-check messages, and more.
```

**Review by romac — APPROVED** [review]
Just a couple suggestions remaining, looks good otherwise! Thanks!

File: CHANGELOG.md (hunk 2 above)
Comment (ljoss17) [reply CHANGELOG.md]: Done in 37744d55428313f595a042df6b3924948cdc5e8c

**Pull request merged**